### PR TITLE
tidy up the travis file for coverage and to make docs pass

### PR DESCRIPTION
**Pull request by Cadair**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -137,28 +137,21 @@ matrix:
            env: SETUP_COMD='test' CONDA_DEPENDENCIES=''
                 PIP_DEPENDENCIES='Cython jinja2 scipy matplotlib mock requests beautifulsoup4 sqlalchemy scikit-image pytest-mock lxml pyyaml pandas hypothesis suds-jurko sphinx-gallery pytest-sugar pytest-rerunfailures sunpy-sphinx-theme glymur pytest-astropy'
 
-
-before_install:
-    # Fix for Python 2 with Numpy 1.14. Should be fixed for 1.14.1
-    - if [[ $TRAVIS_PYTHON_VERSION == 2* ]]; then export NUMPY_VERSION=1.13; fi
-
```
Comment (nabobalis) [inline .travis.yml]: Was this fixed?

File: .travis.yml [hunk 2]
```diff
@@ -75,30 +75,30 @@ matrix:
...
          # Python 3.5 figure tests
          - python: 3.5
            stage: Comprehensive tests
            env: SETUP_CMD='test --figure --coverage' CONDA_DEPENDENCIES=''
 
-         # Python 3.5 non-figure tests
+         # Python 3.5 with astropy 2
          - python: 3.5
            stage: Comprehensive tests
            env: ASTROPY_VERSION='lts'
```
Comment (nabobalis) [inline .travis.yml]: Does LTS work without `astorpy-ci-extras`. I forgot if we moved it to the sunpy channel.

File: .travis.yml (hunk 1 above)
Comment (Cadair) [reply .travis.yml]: well 1.14.1 is out... FOR SCIENCE

File: .travis.yml (hunk 2 above)
Comment (Cadair) [reply .travis.yml]: it should be able to pull it from deafults

File: .travis.yml [hunk 3]
```diff
@@ -34,14 +34,14 @@ python:
 env:
     global:
         - COLUMNS=180
-        - PREVIOUS_NUMPY=1.11.3
+        - PREVIOUS_NUMPY=1.13.3
         - NUMPY_VERSION='stable'
         - ASTROPY_VERSION='stable'
         - MAIN_CMD='python setup.py'
         - SETUP_CMD='test --coverage'
```
Comment (nabobalis) [inline .travis.yml]: Do we not want this to just be `test`? We want to only run coverage on one CI test?

File: .travis.yml (hunk 1 above)
Comment (nabobalis) [reply .travis.yml]: OH I MISSED THAT

File: .travis.yml (hunk 3 above)
Comment (Cadair) [reply .travis.yml]: this is the glory of codecov, we can run it on all of them!

File: .travis.yml (hunk 1 above)
Comment (nabobalis) [reply .travis.yml]: SOMEONE SHOULD TELL ME

File: .travis.yml (hunk 2 above)
Comment (nabobalis) [reply .travis.yml]: OK I CAN REMOVE IT FROM MORE FILES

File: .travis.yml (hunk 3 above)
Comment (nabobalis) [reply .travis.yml]: Oh it just updates the parts we need to?

File: .travis.yml (hunk 1 above)
Comment (Cadair) [reply .travis.yml]: https://about.sibbell.com/

Comment (nabobalis) [reply .travis.yml]: LIKE MAGIC

File: .travis.yml (hunk 3 above)
Comment (dstansby) [reply .travis.yml]: Yep, it supports uploading multiple reports from different tests.

**Pull request merged**